lib/mapObjects/CRewardableObject.cpp

```diff
@@ -83,8 +83,21 @@ void CRewardableObject::onHeroVisit(const CGHeroInstance *h) const
 			if(configuration.info.at(index).reward.removeObject)
 				objectRemovalPossible = true;
 		}
-
 		logGlobal->debug("Visiting object with %d possible rewards", rewards.size());
+
+		if(this->CGObjectInstance::ID == MapObjectBaseID::Type::WARRIORS_TOMB)
+		{
+			for(auto index : rewards)
+			{
+				auto& reword = configuration.info[index].reward;
+				for(auto& bonus : reword.bonuses)
+				{
+					if(bonus.type == BonusType::MORALE && bonus.description.empty())
+						const_cast<std::string&>(bonus.description) = VLC->generaltexth->arraytxt[104]; "\nWarrior Tomb Visited -3";
+				}
+			}
+		}
```

Comment:
This setup part could be filled somewhere else. I have no idea where.

Reply:
Should be in object config, in json. Not hardcoded